### PR TITLE
Renamed IOService to ServerContext

### PR DESCRIPTION
**Pull request by pveentjer**
This is in line with e.g NioNetworking.Context and invocation context.. The context contains
the dependencies for some service.

For EE see: https://github.com/hazelcast/hazelcast-enterprise/pull/3580

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServer.java [hunk 1]
```diff
@@ -65,24 +65,24 @@
...
     private final ILogger logger;
 
     private volatile boolean live;
 
     private final ServerConnectionManager mockConnectionMgr;
     private final AggregateServerConnectionManager mockAggrEndpointManager;
 
-    MockServer(IOService ioService, Node node, TestNodeRegistry testNodeRegistry) {
-        this.ioService = ioService;
+    MockServer(ServerContext serverContext, Node node, TestNodeRegistry testNodeRegistry) {
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServer.java:76]: minor: there is a leftover reference to `NodeIOService` in `MockServer:134`

Comment (pveentjer) [reply hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServer.java:76]: Fixed

File: hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java [hunk 2]
```diff
@@ -262,7 +262,7 @@ private void logClose() {
     }
 
     private Level resolveLogLevelOnClose() {
-        if (!ioService.isActive()) {
+        if (!serverContext.isActive()) {
```
Comment (tkountis) [inline hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java:265]: doesn't read right, active/inactive context?

Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java:265]: That can be fixed.

File: hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionErrorHandler.java [hunk 3]
```diff
@@ -34,10 +34,10 @@
 
     TcpServerConnectionErrorHandler(TcpServerConnectionManager connectionManager, Address endPoint) {
         this.endPoint = endPoint;
-        this.ioService = connectionManager.getServer().getIoService();
-        this.minInterval = ioService.getConnectionMonitorInterval();
-        this.maxFaults = ioService.getConnectionMonitorMaxFaults();
-        this.logger = ioService.getLoggingService().getLogger(getClass());
+        this.serverContext = connectionManager.getServer().getContext();
+        this.minInterval = serverContext.getConnectionMonitorInterval();
```
Comment (tkountis) [inline hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionErrorHandler.java:38]: np: getters make no sense here i guess.

File: hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java (hunk 2 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java:265]:
```
  @Override
    public boolean isNodeActive() {
        return node.getState() != NodeState.SHUT_DOWN;
    }
```

**Pull request merged**